src/Controller/CRUDController.php

```diff
@@ -647,16 +751,16 @@ public function historyAction(Request $request)
 
         $this->admin->checkAccess('history', $object);
 
-        $manager = $this->get('sonata.admin.audit.manager');
-
-        if (!$manager->hasReader($this->admin->getClass())) {
-            throw $this->createNotFoundException(sprintf(
-                'unable to find the audit reader for class : %s',
-                $this->admin->getClass()
-            ));
+        if (!$this->auditManager->hasReader($this->admin->getClass())) {
+            throw $this->createNotFoundException(
+                sprintf(
+                    'unable to find the audit reader for class : %s',
```

Comment:
It's a good opportunity to fix this message:
```suggestion
                    'Unable to find the audit reader for class %s.',
```